app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java

```diff
@@ -435,13 +436,15 @@ private void scrollToCurrent(@NonNull List<Lecture> lectures, int day, int heigh
                 }
             }
 
-            for (Lecture l : lectures) {
-                if (l.day == day && l.startTime <= time && l.startTime + l.duration > time) {
-                    if (col == -1 || col >= 0 && l.roomIndex == col) {
-                        MyApp.LogDebug(LOG_TAG, l.title);
-                        MyApp.LogDebug(LOG_TAG, time + " " + l.startTime + "/" + l.duration);
-                        scrollAmount -= ((time - l.startTime) / 5) * height;
-                        time = l.startTime;
+            List<RoomData> roomDataList = scheduleData.getRoomDataList();
+            if (col >= 0 && col < roomDataList.size()) {
+                RoomData roomData = roomDataList.get(col);
+                for (Lecture lecture : roomData.getLectures()) {
+                    if (lecture.startTime <= time && lecture.startTime + lecture.duration > time) {
+                        MyApp.LogDebug(LOG_TAG, lecture.title);
+                        MyApp.LogDebug(LOG_TAG, time + " " + lecture.startTime + "/" + lecture.duration);
+                        scrollAmount -= ((time - lecture.startTime) / 5) * height;
```

Comment:
Maybe the magic 5 is the same as LayoutCalculator.DIVISOR? I even think you could call LayoutCalculator.calculateDisplayDistance(time-lecture.startTime) which would semantically be correct